### PR TITLE
Antenna calculator: verify math, fix bugs, add tests

### PR DESCRIPTION
**Pull request by lrnselfreliance**
## Summary
- Extract pure antenna math from the dipole calculator (ARRL constants: full-wave `936/f`, half-wave `468/f`, quarter-wave `234/f`, 5/8-wave `585/f`; free-space λ via exact `c`).
- Fix feet/inches display that could show **"N ft 12 in"** instead of rolling over to the next foot.
- Make wavelength input bidirectional so editing wavelength recalculates frequency and antenna lengths.
- Add 50 unit tests covering formulas, band sanity, unit consistency, end-effect vs free space, and edge cases.

## Test plan
- [x] `cd app && CI=true npm test -- --watchAll=false src/components/calculators/HamCalculators.test.js` (50 passed)
- [ ] Open Calculators → Antenna; confirm default 144.2 MHz half-wave shows ~3 ft 3 in total / 1 ft 7 in per leg
- [ ] Change MHz (e.g. 14.2) and confirm lengths update and wavelength field updates
- [ ] Edit wavelength field and confirm MHz + length tables update
- [ ] Spot-check ¼ and ⅝ tables against 234/f and 585/f

### CODE REVIEW

File: app/src/components/calculators/HamCalculators.js [hunk 1]
```diff
@@ -5,69 +5,231 @@ import Grid from "semantic-ui-react/dist/commonjs/collections/Grid";
...
     const handleWaveChange = (e, {value}) => {
         setWaveInputValue(value);
-        setWave(value && parseFloat(value) > 0 ? parseFloat(value) : 0);
+        const parsed = parseFloat(value);
+        if (value && parsed > 0) {
+            const newMhz = frequencyMhzFromWavelengthMeters(parsed);
+            setMhz(newMhz);
+            setMhzInputValue(roundDigits(newMhz));
+            localStorage.setItem(mhzStorageKey, String(roundDigits(newMhz)));
+        }
```
[the remark this answers is not in the document]
Comment (lrnselfreliance) [reply app/src/components/calculators/HamCalculators.js]: Addressed in follow-up: cleared/zero/non-finite wavelength now zeros MHz, clears the MHz field, and resets localStorage so tables do not keep the previous calculation.

[the remark this answers is not in the document]
Comment (lrnselfreliance) [reply app/src/components/calculators/HamCalculators.js]: Addressed in follow-up: require `Number.isFinite(parsed) && parsed > 0`, and also guard a null conversion result so overflow/`NaN` is never written to state or localStorage. Applied the same finite check on the MHz handler.
